cns/restserver/nodesubnet_test.go

```diff
@@ -0,0 +1,78 @@
+package restserver_test
+
+import (
+	"context"
+	"testing"
+
+	"github.com/Azure/azure-container-networking/cns"
+	"github.com/Azure/azure-container-networking/cns/logger"
+	"github.com/Azure/azure-container-networking/cns/restserver"
+)
+
+// Mock implementation of PodInfoByIPProvider
+type MockPodInfoByIPProvider struct{}
+
+func (m *MockPodInfoByIPProvider) PodInfoByIP() (res map[string]cns.PodInfo, err error) {
+	return res, nil
+}
+
+// Mock implementation of CNIConflistGenerator
+type MockCNIConflistGenerator struct {
+	GenerateCalled chan bool
+}
+
+func (m *MockCNIConflistGenerator) Generate() error {
+	m.GenerateCalled <- true
```

Comment:
Usually for a latch like this, it's more common to do `chan struct{}`. Instead of sending a value on it, you just close it (`close(m.GenerateCalled)`). The difference is that sending a value will wake up one waiting goroutine (once this goroutine is able to perform the write), where closing will wake up all of them. This is usually what you want when writing tests like this.